addons/account/static/src/components/dynamic_selection/dynamic_selection.js

```diff
@@ -0,0 +1,46 @@
+/** @odoo-module **/
+
+import { registry } from "@web/core/registry";
+import { SelectionField, selectionField } from "@web/views/fields/selection/selection_field";
+import { standardFieldProps } from "@web/views/fields/standard_field_props";
+
+export class DynamicSelectionField extends SelectionField {
+
+    static props = {
+        ...standardFieldProps,
```

Comment:
Unless I'm mistaken, I guess you could inherit the props as well
```suggestion
        ...selectionField.props,
```